api/modules/publishedFiles/cache.js

```diff
@@ -0,0 +1,169 @@
+"use strict";
+
+const Promise = require(`bluebird`);
```

Comment:
not exactly. bluebird has some pretty useful promise based functions that help us handle collections. Once we switch to node 6, I would use generators instead.